scipy/sparse/sparsetools/coo.h

```diff
@@ -146,6 +193,40 @@ void coo_matvec(const npy_int64 nnz,
 }
 
 
+/*
+ * Input Arguments:
+ *   npy_int64  nnz         - number of nonzeros in A
+ *   npy_int64 num_dims     - number of dimensions
+ *   I  shape[num_dims]     - shape of A
+ *   I  Aijk[nnz * num_dims]- coords for nonzeros in A
+ *   T  Ax[nnz]             - nonzero values in A
+ *   T  Xx[n_col]           - input vector
+ *
+ * Output Arguments:
+ *   T  Yx[n_row]     - output vector
+ * 
+ */
+template <class I, class T>
+void coo_matvec_nd(const npy_int64 nnz,
+                const npy_int64 num_dims,
+                const I shape[],
```

Comment:
Same comment here: we could pass `strides[]` instead of `shape[]`.

Reply:
Addressed at 54d60d6.